spec/utils_spec.lua

```diff
@@ -1027,13 +1027,55 @@ describe("utils", function()
 			local name = undername .. "_" .. basename
 
 			local undernameFromName = utils.undernameForName(name)
-			assert.are.same(undernameFromName, undername)
+			assert.are.same(undername, undernameFromName)
 		end)
 
 		it("should return nil for a name with no undername", function()
 			local basename = "basename"
 			local undernameFromName = utils.undernameForName(basename)
-			assert.are.same(undernameFromName, nil)
+			assert.are.same(nil, undernameFromName)
+		end)
+
+		it("should get the undername name for a name with an undername and base name with dashes", function()
+			local undername = "test"
+			local basename = "base-name-with-dashes"
+			local name = undername .. "_" .. basename
+
+			local undernameFromName = utils.undernameForName(name)
+			assert.are.same(undername, undernameFromName)
+		end)
+
+		it("should return nil for a base name with dashes but no undername", function()
+			local basename = "base-name-with-dashes"
+			local undernameFromName = utils.undernameForName(basename)
+			assert.are.same(nil, undernameFromName)
+		end)
+
+		it("should handle complex undername with base name containing dashes", function()
```

Comment:
Nit: "should handle" is not descriptive enough of the intended behavior so it's harder for a reader to understand whether the test is implemented appropriately. The tests above do a better job describing the expected result.